pkg/etcdcli/health.go

```diff
@@ -43,34 +43,21 @@ type memberHealth []healthCheck
 
 func GetMemberHealth(ctx context.Context, etcdMembers []*etcdserverpb.Member) memberHealth {
 	memberHealth := memberHealth{}
+	resChan := make(chan healthCheck, 1)
 	for _, member := range etcdMembers {
 		if !HasStarted(member) {
 			memberHealth = append(memberHealth, healthCheck{Member: member, Healthy: false})
 			continue
 		}
 
-		const defaultTimeout = 30 * time.Second
-		resChan := make(chan healthCheck, 1)
 		go func() {
-			ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeout)
-			defer cancel()
-
-			resChan <- checkSingleMemberHealth(ctxTimeout, member)
-			// closing here to avoid late replies to panic on resChan,
-			// the result will be considered a timeout anyway
-			close(resChan)
+			resChan <- checkSingleMemberHealth(ctx, member)
```

Comment:
- what if the caller has not set the ctx timeout,  please see https://github.com/openshift/cluster-etcd-operator/blob/0a00f2ac2d578114cea7164dbdba4c98ce06c69e/pkg/etcdcli/etcdcli.go#L415-L427
-  and in extreme cases, this call would be blocked until the linux system releases this abnormal connection(which maybe 15 minutes)
- maybe u could set the caller's ctx timeout as 30 * time.Second

Reply:
Hey @lance5890 , unless I'm misunderstanding your comment here my PR updates that specific call site to have a timeout of 30s as well. It was the only one I'm the entire repo I found that didn't.

Reply:
o, my mistake,  it looks good